docs/HACKING.md

```diff
@@ -112,47 +142,293 @@ Additional flags that can be passed to `hack/run-ci-e2e-test.sh` are
-instance(s) information must be provided to the e2e test suite through the `WINDOWS_INSTANCES_DATA` environment
-variable.
+To know which windows version your release uses, check the defaultSKU in
+[azure.go](<https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/providers/azure/azure.go#L21>)
+for the WMCO release you’re testing with.
+To know which base image your release uses, check the
+[getWindowsServerContainerImage()](<https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/network_test.go#L460>)
+function in network_test.go for the WMCO release you’re testing with.
 
-To deploy machines using the [infrastructure providers supported by WMCO](wmco-prerequisites.md#supported-cloud-providers-based-on-okdocp-version-and-wmco-version),
-refer to the specific provider documentation. See Windows instance [pre-requisites](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/byoh-instance-pre-requisites.md).
+##### Check the operating system on the cluster
 
-The information you need to collect from each Windows instance is:
-- the internal IP address
-- the Windows administrator username
+To be sure that your deployment’s base image corresponds with the cluster’s OS version, you should check what version
+of OS your cluster is running.
 
-Export `WINDOWS_INSTANCES_DATA` as an environment variable with the corresponding [windows-instances ConfigMap](https://github.com/openshift/windows-machine-config-operator#adding-instances)
-data section, and a new `windows-instances` ConfigMap will be created during the execution of the e2e test suite.
+##### Edit your deployment
 
-For example:
-```shell
+Now that you know what operating system is running, you have to find the corresponding base image to set your
+deployment to. This is an incomplete table of which windows version corresponds with which powershell base image.
+
+| Windows Version                 | Base Image version                                   |
+|---------------------------------|------------------------------------------------------|
+| 2019-Datacenter-with-Containers | mcr.microsoft.com/powershell:lts-nanoserver-1809     |
+| 2022-datacenter-smalldisk       | mcr.microsoft.com/powershell:lts-nanoserver-ltsc2022 |
+| datacenter-core-20h2-with-containers-smalldisk | mcr.microsoft.com/powershell:lts-nanoserver-20h2|
+
+If you are using the example deployment, a sed command can update the base image for you.
+
+```shell script
+export BASE_IMAGE=powershell:lts-nanoserver-ltsc2022
+sed -i 's/image:.*$/image: "mcr.microsoft.com\/'$BASE_IMAGE'"/' deployment.yaml
+```
+
+##### Set the powershell executable
+
+Different base images have different executables for powershell. Since WMCO uses the powershell nanoserver images, the
+command should be set to pwsh.exe.
+Change the executable to pwsh.exe by running
+
+```shell script
+sed -i "s/powershell.exe/pwsh.exe/" deployment.yaml
+```
```

Comment:
I'm not sure all the workload and container image info is best lived in HACKING.md. Could this be reframed to be included in TROUBLESHOOTING.md instead? or it's own doc?

Reply:
I think you're right about this, but since deploying a custom workload seems to be one of the more common things you should know how to do, we should have instructions that take you through the whole process, and since the hacking.md is for building and testing, running a custom workload is a type of test that you should be familiar with. If we made it its own doc, I don't know why we would link out to it instead of keeping it in the HACKING.md where people would expect to find it